### PR TITLE
[WIP] Add comments to the authentications table

### PR DESCRIPTION
**Pull request by djberg96**
In my continuing effort to understand things better by documenting them, this PR adds comments for the columns of the `authentications` table.

WIP for now as there are many columns I know nothing about, and some of the current ones I'm guessing on.

### CODE REVIEW

File: db/migrate/20190219180039_add_comments_to_authentications_table.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+class AddCommentsToAuthenticationsTable < ActiveRecord::Migration[5.0]
+  def up
+    change_table_comment :authentications, "Authentication information for various resources."
+
+    change_column_comment :authentications, :id, "[builtin] The internal record ID. This is the primary key."
+    change_column_comment :authentications, :name, "Symbolic name for the authentication"
+    change_column_comment :authentications, :authtype, "The authentication type or role, e.g. ssh_keypair, default, amqp, etc."
```
Comment (agrare) [inline db/migrate/20190219180039_add_comments_to_authentications_table.rb]: > The authentication type or role

Not a comment on the comment ;) But this is my biggest source of confusion.  This is as accurate as I think we can be, it is either the type of authentication (e.g. sometimes it is ssh_keypair or bearer token) and sometimes it is the role (e.g. hawkular, prometheus, metrics).

I would have expected the the type would have been encoded in the STI type and authtype would just be the role.

**Comment by djberg96** [comment]
@jvlcek I don't suppose you could speak to any of these, could you? The ldap columns perhaps?

File: db/migrate/20190219180039_add_comments_to_authentications_table.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    change_column_comment :authentications, :service_account
+    change_column_comment :authentications, :challenge
+    change_column_comment :authentications, :login
+    change_column_comment :authentications, :public_key, "An SSH public key."
+    change_column_comment :authentications, :htpassd_users
+    change_column_comment :authentications, :ldap_id
+    change_column_comment :authentications, :ldap_email
+    change_column_comment :authentications, :ldap_name
+    change_column_comment :authentications, :ldap_preferred_user_name
+    change_column_comment :authentications, :ldap_bind_dn
```
Comment (jvlcek) [inline db/migrate/20190219180039_add_comments_to_authentications_table.rb]: bind dn - Is the LDAP identity by which LDAP operations are performed.

File: db/migrate/20190219180039_add_comments_to_authentications_table.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    change_column_comment :authentications, :ldap_insecure
```
Comment (jvlcek) [inline db/migrate/20190219180039_add_comments_to_authentications_table.rb]: I am not sure but I assume this is likely a boolean indicating LDAP or LDAPS

File: db/migrate/20190219180039_add_comments_to_authentications_table.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    change_column_comment :authentications, :ldap_url
```
Comment (jvlcek) [inline db/migrate/20190219180039_add_comments_to_authentications_table.rb]: Most likely the URL to the LDAP server.

File: db/migrate/20190219180039_add_comments_to_authentications_table.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
+class AddCommentsToAuthenticationsTable < ActiveRecord::Migration[5.0]
+  def up
+    change_table_comment :authentications, "Authentication information for various resources."
+
+    change_column_comment :authentications, :id, "[builtin] The internal record ID. This is the primary key."
+    change_column_comment :authentications, :name, "Symbolic name for the authentication"
+    change_column_comment :authentications, :authtype, "The authentication type or role, e.g. ssh_keypair, default, amqp, etc."
+    change_column_comment :authentications, :userid, "The userid used in a userid + password authentication approach."
```
Comment (jvlcek) [inline db/migrate/20190219180039_add_comments_to_authentications_table.rb]: The userid is the unique string identifier used in the MiQ database to identify a user record. The format of the userid varies depending on authentication mode but for all external authentication modes it is UPN format.

File: db/migrate/20190219180039_add_comments_to_authentications_table.rb (hunk 4 above)
Comment (jvlcek) [inline db/migrate/20190219180039_add_comments_to_authentications_table.rb]: @djberg96 I tried to shed some light on some of these but I am not sure on some. If you would like to schedule a meeting I'd be glad to  research them together.
Joe

**Pull request closed without merging**